src/diff_py_app.cc

```diff
@@ -0,0 +1,190 @@
+#include "diff_py_app.h"
+#include "py_wrappers.h"
+#include "opensn/python/lib/console.h"
+#include "opensn/python/lib/py_wrappers.h"
+#include "opensn/framework/logging/log.h"
+#include "opensn/framework/utils/utils.h"
+#include "opensn/framework/utils/timer.h"
+#include "opensn/framework/runtime.h"
+#include "caliper/cali.h"
+#include "cxxopts/cxxopts.h"
+#include "petsc.h"
+#include <string>
+
+using namespace opensn;
+namespace py = pybind11;
+
+namespace diffpy
+{
+
+DiffApp::DiffApp(const mpi::Communicator& comm) : allow_petsc_error_handler_(false)
+{
+  opensn::mpi_comm = comm;
+
+  py::module sys = py::module::import("sys");
+  py::exec("rank = " + std::to_string(comm.rank()));
+  py::exec("size = " + std::to_string(comm.size()));
+  py::exec("opensn_console = True");
+
+  opensnpy::console.BindBarrier(comm);
+
+  opensnpy::console.BindModule(WrapYlm);
+  opensnpy::console.BindModule(WrapVector3);
+  opensnpy::console.BindModule(WrapFunctors);
+
+  opensnpy::console.BindModule(WrapQuadraturePointPhiTheta);
+  opensnpy::console.BindModule(WrapQuadrature);
+  opensnpy::console.BindModule(WrapProductQuadrature);
+  opensnpy::console.BindModule(WrapCurvilinearQuadrature);
+  opensnpy::console.BindModule(WrapSLDFESQuadrature);
+
+  opensnpy::console.BindModule(WrapMesh);
+  opensnpy::console.BindModule(WrapMeshGenerator);
+  opensnpy::console.BindModule(WrapGraphPartitioner);
+
+  opensnpy::console.BindModule(WrapLogicalVolume);
+
+  opensnpy::console.BindModule(WrapPointSource);
+  opensnpy::console.BindModule(WrapVolumetricSource);
+
+  opensnpy::console.BindModule(WrapMultiGroupXS);
+
+  opensnpy::console.BindModule(WrapFieldFunction);
+  opensnpy::console.BindModule(WrapFieldFunctionGridBased);
+  opensnpy::console.BindModule(WrapFieldFunctionInterpolation);
+
+  opensnpy::console.BindModule(WrapResEval);
+
+  opensnpy::console.BindModule(WrapProblem);
+  opensnpy::console.BindModule(WrapSolver);
+  opensnpy::console.BindModule(WrapLBS);
+  opensnpy::console.BindModule(WrapSteadyState);
+  opensnpy::console.BindModule(WrapNLKEigen);
+  opensnpy::console.BindModule(WrapPIteration);
+  opensnpy::console.BindModule(WrapPRK);
+
+  opensnpy::console.BindModule(WrapDiffusion);
+
+  opensnpy::console.BindModule(WrapPostProcessor);
+  opensnpy::console.BindModule(WrapPrinter);
```

Comment:
The way this should probably work is that  we call the binding function from OpenSn (rather than copying that). Then, add the application wrappers like `WrapDiffusion`